commit 58f5a7151efe2a3039b097883e6ba9fb24e42536
Author: Alexey Galakhov <agalakhov@gmail.com>
Date:   Wed Jan 30 15:35:22 2013 +0600

    ASoC: samsung: Fix compilation error on S3C2440
    
    The commit a08485d8 ("ASoC: Samsung: Do not register samsung audio dma
    device as pdev") introduced compilation error.
    
    Combination of CONFIG_CPU_S3C2440 and CONFIG_SND_S3C24XX_I2S
    caused undefined symbols 's3c_i2sv2_register_dai' and 's3c2412_i2s_dai'
    in sound/soc/samsung/s3c24xx-i2s.c.
    
    This patch fixes the problem and makes S3C2440 I2S usable again. It does
    not affect S3C2412 (aka I2S-v2).
    
    Signed-off-by: Alexey Galakhov <agalakhov@gmail.com>
    Signed-off-by: Mark Brown <broonie@opensource.wolfsonmicro.com>

diff --git a/sound/soc/samsung/s3c24xx-i2s.c b/sound/soc/samsung/s3c24xx-i2s.c
index ee10e87..13f6dd1 100644
--- a/sound/soc/samsung/s3c24xx-i2s.c
+++ b/sound/soc/samsung/s3c24xx-i2s.c
@@ -469,7 +469,7 @@ static int s3c24xx_iis_dev_probe(struct platform_device *pdev)
 {
 	int ret = 0;
 
-	ret = s3c_i2sv2_register_dai(&pdev->dev, -1, &s3c2412_i2s_dai);
+	ret = snd_soc_register_dai(&pdev->dev, &s3c24xx_i2s_dai);
 	if (ret) {
 		pr_err("failed to register the dai\n");
 		return ret;
